### PR TITLE
feat: added K8s CronJob implementation

### LINKED ISSUE #12: [FR] Implement container internal cronjob via k8s CronJob
Right now we still use the crond started in the container itself to executed jobs periodically.

This feature request is about the implementation of this mechanism via the CronJob implemented in k8s: https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/

Note: We need to find a way to execute the job in the container without the triggering any updated of the oC system.

### PR DESCRIPTION
**Pull request by d7oc**
This PR implements https://github.com/owncloud-docker/helm-charts/issues/12.

### CODE REVIEW

**Comment by d7oc** [comment]
@xoxys How should we deal with [`cronLog`](https://github.com/owncloud-docker/helm-charts/blob/60499d8044f83c80cd2f66e7c36b8aad21c3d4e5/charts/owncloud/values.yaml#L69). This setting is no more needed with this PR as the logs of the CronJob execution will be found in K8s.

**Comment by xoxys** [comment]
Drop it I would say.

**Comment by xoxys** [comment]
I'm not that convinced by the huge include block for the env vars. Wouldn't be using a ConfigMap cleaner if we need to share all env vars with other containers?

**Comment by d7oc** [comment]
> I'm not that convinced by the huge include block for the env vars. Wouldn't be using a ConfigMap cleaner if we need to share all env vars with other containers?

We could switch to the ConfigMap (or Secret in here), but the pure amount will still be the same. I'm also not quite sure if there won't be any conflict between the still existing `overwrite.config.php` and the new file which we will place there build from the config map. I see parameters like those https://github.com/owncloud-docker/base/blob/62d4e7ff12f79fd3a02b17ab6fdca432605ca870/latest/overlay/etc/templates/config.php#L25-L31 which would be set empty as there is no check upfront. So there might be a race condition if I'm right. What we might try is to place the config map "on top" the `overwrite.config.php` with an overlay mount. I'm just not sure if this works. Needs a test.

So in a nutshell the template approach sounds simpler to me.

**Comment by xoxys** [comment]
That's not what I was talking about. The idea was to just use a ConfigMap to properly share the env vars by using `envFrom.configMapRef`, see https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables

**Comment by d7oc** [comment]
Ah this. Ok this might work. Will give it a try.

**Comment by d7oc** [comment]
Done, also tested working on a local OpenShift cluster, please recheck.

**Comment by xoxys** [comment]
How is `crondEnabled` intended to be used? Right now, cronjobs can't be disabled, right?

File: charts/owncloud/templates/cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          containers:
+          - name: {{ .Chart.Name }}-cronjob
+            securityContext:
+              {{- toYaml .Values.securityContext | nindent 14 }}
+            image: {{ template "owncloud.image" . }}
+            imagePullPolicy: {{ .Values.image.pullPolicy }}
+            envFrom:
+              - configMapRef:
+                  name: owncloud-env-config
+            command: ['/usr/bin/entrypoint', "/usr/bin/owncloud", "/usr/bin/cronjob"]
```
Comment (xoxys) [inline charts/owncloud/templates/cronjob.yaml]: Mixed single/double quotes.
How is this intended to work? Why does it need to run the entire startup scripts instead of just `/usr/bin/cronjob`?

Comment (d7oc) [reply charts/owncloud/templates/cronjob.yaml]: > Mixed single/double quotes.

Addressed in https://github.com/owncloud-docker/helm-charts/pull/45/commits/3759f8e3ea4b9c762f19eef22d96633371350e23. Worked nevertheless.

> How is this intended to work? Why does it need to run the entire startup scripts instead of just `/usr/bin/cronjob`?

This is the only way it works. This goes back to a discussion we had in July 2022. The `/usr/bin/entrypoint` script doesn't initialise ownCloud correctly thus you cannot run any `occ`-related commands as it will complain that ownCloud is not installed which is due to symlinks and env variables being missing. Back then it was decided, that fixing `/usr/bin/entrypoint` is too risky instead only a documentation should be added which resulted in https://github.com/owncloud/docs-server/pull/507.

**Comment by d7oc** [comment]
> How is `crondEnabled` intended to be used? Right now, cronjobs can't be disabled, right?

You're right, this was missing, good catch. Added in https://github.com/owncloud-docker/helm-charts/pull/45/commits/c85c92d74dbbfb05adb804fdd272aaa76c8aada2.

**Review by xoxys — APPROVED** [review]
